lib/features/down_by_maintenance/domain/services/check_server_status_service.dart

```diff
@@ -0,0 +1,17 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter_template/features/down_by_maintenance/domain/entities/check_result.dart';
+
+/// Interface for checking server status.
+abstract interface class ICheckServerStatusService {
```

Comment:
It should be a repository. Rename it and its implementations, please.

Reply:
\+